clouddriver-google-common/src/main/groovy/com/netflix/spinnaker/clouddriver/googlecommon/batch/GoogleBatchRequest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.googlecommon.batch;
+
+import com.google.api.client.googleapis.batch.BatchRequest;
+import com.google.api.client.googleapis.batch.json.JsonBatchCallback;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.ComputeRequest;
+import com.google.common.collect.Iterables;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Helper class for sending batch requests to GCE.
+ */
+@Slf4j
+public class GoogleBatchRequest {
+
+  private static final Integer MAX_BATCH_SIZE = 100; // Platform specified max to not overwhelm batch backends.
+  private static final Integer DEFAULT_CONNECT_TIMEOUT_MILLIS = 2 * 60000;
+  private static final Integer DEFAULT_READ_TIMEOUT_MILLIS = 2 * 60000;
+
+  private ArrayList<QueuedRequest> queuedRequests;
```

Comment:
You can make this a `List`, unless we can only use `ArrayList` as the concrete type.